laravel/resources/views/pages/event/view.blade.php

```diff
@@ -4,11 +4,16 @@
     <section class="event" data-id="{{ $event->id }}">
         <div class="pull-right relative" style="z-index: 1">
             @can('publish-event')
-                @if($event->published_at === null)
-                    <a href="/event/{{ $event->id }}/publish" class="btn btn-primary">Publish Event</a>
-                @else 
-                    <a href="/event/{{ $event->id }}/publish" class="btn btn-primary">Unpublish Event</a>
-                @endif
+                <span style="display:inline-block">
+                    {!! Form::open(['url' => "/event/{$event->id}/publish", 'method' => 'post']) !!}
+                    <?php
+                        $unpublished = ($event->published_at === null);
+                        $publish_text = ($unpublished) ? 'Publish Event' : 'Unpublish Event';
+                    ?>
```

Comment:
why not handle this in the controller? seems a bit weird and hacky to do it like this.

Reply:
it does but any logic that determines view-based information, formatting or content representing the state should be kept in the front end. we just don't have a neat front-end to transform the backend data more beautifully.